lib/wanda/execution/evaluation.ex

```diff
@@ -99,14 +111,15 @@ defmodule Wanda.Execution.Evaluation do
 
   defp eval_expectation(
          %Expectation{name: name, type: type, expression: expression},
-         facts
+         facts,
+         values
        ) do
-    scoped_facts =
-      Enum.reduce(facts, %{}, fn %Fact{name: name, value: value}, acc ->
-        put_in(acc, [name], value)
-      end)
+    evaluation_scope = %{
```

Comment:
In order to have some consistency with the evaluation scope, I would create a generic function like
`namespace_scope`. Something like:
```
def namespace_scope(facts, values, env) do
    scoped_facts =
      Enum.reduce(facts, %{}, fn %Fact{name: name, value: value}, acc ->
        put_in(acc, [name], value)
      end)

    scoped_values =
      Enum.reduce(values, %{}, fn %Value{name: name, value: value}, acc ->
        put_in(acc, [name], value)
      end)

    %{"facts" => scoped_facts, "values" => scoped_values, "env" => env}
  end
```

This way, we can create the evaluation scope more or less the same way in all the evaluations.

For example, above, we could have something like: 
```
values = Enum.map(values, &eval_value(&1, namespace_scope(facts, [], env)))
# and for the expectations
Enum.map(expectations, &eval_expectation(&1, namespace_scope(facts, values, env)))
```

Reply:
We could even have a struct to make it more strict, with some function to convert it to map

Reply:
we could use the builder pattern, something like:
```elixir
%{} |> add_scope("facts", facts) |> add_scope("values", values)
```

where `add_scope/3` is a function that takes a map as first parameter and does the transformation I proposed here: https://github.com/trento-project/wanda/pull/46#discussion_r989742219

Reply:
Nice @fabriziosestito , I like this approach more. Only 1 new function